decoder/completion_resolve.go

```diff
@@ -0,0 +1,20 @@
+package decoder
+
+import (
+	"context"
+	"errors"
+)
+
+// ResolveCandidate tries to gather more information for a candidate,
+// by checking for a resolve hook and executing it.
+func (d *Decoder) ResolveCandidate(ctx context.Context, unresolvedCandidate UnresolvedCandidate) (*ResolvedCandidate, error) {
+	if unresolvedCandidate.ResolveHook == nil {
+		return nil, errors.New("missing resolve hook")
```

Comment:
```suggestion
		return nil, nil
```
We could of course ignore that error downstream in LS or assume extra check for `nil` there, but I'm not sure there's any benefit of raising that error in the first place? Typically we'd return errors to make testing easier, but I think this is testable even without the error?